pmd-core/src/main/java/net/sourceforge/pmd/RuleSet.java

```diff
@@ -499,11 +499,11 @@ public void apply(List<? extends Node> acuList, RuleContext ctx) {
                         rule.apply(acuList, ctx);
                     } catch (RuntimeException e) {
                         if (ctx.isIgnoreExceptions()) {
-                            ctx.getReport().addError(new Report.ProcessingError(e, ctx.getSourceCodeFilename()));
+                            ctx.getReport().addError(new Report.ProcessingError(e, String.valueOf(ctx.getSourceCodeFile())));
```

Comment:
isn't this equivalent to calling `ctx.getSourceCodeFile().toString()`, or more explicitly `ctx.getSourceCodeFile().getPath()`?

Reply:
This way, we might avoid a NPE, in case ctx.getSourceCodeFile() is null.

Reply:
good call, shouldn't happen, but better be safe